official/utils/arg_parsers/parsers.py

```diff
@@ -142,16 +142,29 @@ def __init__(self, add_help=False, data_dir=True, model_dir=True,
     if batch_size:
       self.add_argument(
           "--batch_size", "-bs", type=int, default=32,
-          help="[default: %(default)s] Batch size for training and evaluation.",
+          help="[default: %(default)s] Global batch size for training and "
+               "evaluation.",
           metavar="<BS>"
       )
 
+    assert not (multi_gpu and num_gpu)
+
     if multi_gpu:
       self.add_argument(
           "--multi_gpu", action="store_true",
           help="If set, run across all available GPUs."
       )
 
+    if num_gpu:
+      self.add_argument(
+          "--num_gpus", "-ng",
+          type=int,
+          default=1 if tf.test.is_built_with_cuda() else 0,
```

Comment:
Can you add a test to make sure this default gets set correctly in the arg parsers test? I think there's a way to force one mode or the other... if not, have you at least confirmed that this does work correctly on GPU/CPU?

Reply:
I tested manually and confirmed. Probably not worth the effort to make a formal test simply because at that point it's more of a test of tf.test than the model garden.